app/forms/fields/decimal_field_with_separator.py

```diff
@@ -22,9 +23,13 @@ def __init__(self, **kwargs):
 
     def process_formdata(self, valuelist):
         if valuelist:
+            if valuelist[0] == "NaN":
+                raise ValidationError(error_messages["INVALID_NUMBER"])
```

Comment:
This shouldn't be done here, this is a job for the validators. This should be done in our `NumberRange` validator.

You need to update it to:
```py
value: Union[int, Decimal] = field.data
if value is not None and not math.isnan(value):
```

This matches how wtform official handles it. https://github.com/wtforms/wtforms/pull/548/files

Reply:
The problem here is, by the time we use `NumberRange` validator the form field.data is already transformed to `None` from the original `NaN`, so I am not sure what is the solution here.

Reply:
But that is not an issue is it because they are asked to enter a valid number...?

Reply:
Ok, had to slightly modify your suggestion to make it work, thanks.